tests/leverage_brackets.py

```diff
@@ -0,0 +1,1215 @@
+def leverage_brackets():
```

Comment:
Can this potentially use the "dry-run" json file?

Reply:
It could, the test ends up being just a copy and paste of the actual code though, notice in here they're ints, but in the dry run json they're strings, and they're only strings in the dry-run json because I didn't want the code to differ too much between dry and live run

Reply:
I had it using that json in an earlier commit, but then thought it was dumb because it was almost a copy and paste of the method it was testing

Reply:
my problem (a bit) is that this is 1200 lines of json - which we now have twice ...

maybe for the test, reduce it to 3-4 entries?